Adds filters for datetime functions.

mattcoley opened the pull request:
Adds `plus_time`, `minus_time`, and `between_times` filters to operate on datetime objects/timestamp. Examples: `date|plus_time('1', 'days')` will give you a date 1 day in the future; `begin|between_times(end, 'minutes')` will give you the number of minutes between those two dates.

I also added a `strtotime` to convert a string and datetime format to a datetime object. For example, `"2018-07-14T14:31:30+0530"|strtotime("yyyy-MM-dd'T'HH:mm:ssZ")|unixtimestamp` will output `1531558890000`

Fixes https://github.com/HubSpot/jinjava/issues/56

boulter commented:
Do these have any precedent in Jinja? I would prefer to keep it as compatible as possible. If not, perhaps these could filters could be named with a little more descriptively like `plus_time`, `minus_time` and `between_times`.

mattcoley commented:
The recommended way in jinja is with python commands. We could encourage developers to use Java methods like `plusDays()` but those seem like the only options.

mattcoley commented:
I changed this to support all [ChronoUnit](https://docs.oracle.com/javase/8/docs/api/java/time/temporal/ChronoUnit.html) enum values and added a link at the top of each class.
